ofono/bridge.py

```diff
@@ -0,0 +1,135 @@
+        else:
+            logger.info("Powering up modem...")
+            try:
+                self._bus.SetProperty("Powered", pydbus.Variant('b', True))
+                sleep(2)  # Let the modem some time to initialize
+            except Exception:
+                logger.error("Couldn't power up the modem !")
+
+    def power_off(self):
+        self._bus.SetProperty("Powered", pydbus.Variant('b', False))
+
+    def send_sms(self, to, content):  # todo : untested
+        self.message_manager.SendMessage(to, content)
+        print("Sending", to, content)
+        ConversationManager().on_new_message_sent(to, content)
+
+    @staticmethod
+    def on_message_received(message, details, path=None, interface=None):  # todo : untested
+        logger.info("Got message with path {}".format(path))
+        ConversationManager().on_new_message_received(message, details)
+
+    def _listen_messages(self):
+        logger.info("Connecting to dbus callbacks")
+        self.message_manager.IncomingMessage.connect(self.on_message_received)
+        self.message_manager.ImmediateMessage.connect(self.on_message_received)
+
+    def _init_messages(self):
+        self.message_manager.SetProperty("UseDeliveryReports", pydbus.Variant('b', True))
+
+
+class ConversationManager(Singleton):
+    """
+    Singleton dedicated to conversations. Logs every message sent and received in a flat file
+    for any given phone number
+    """
+
+    def __init__(self):
+        super(ConversationManager, self).__init__()
+        self.folder = os.path.expanduser("~/.phone/sms/")  # todo: store as a constant somewhere
+        if not os.path.exists(self.folder):
+            os.mkdir(self.folder)
+
+    def on_new_message_sent(self, to, content):
+        logger.info("Sent message to '{}'".format(to))
+        self._write_log(to, self._format_log(content, from_me=True))
+
+    def on_new_message_received(self, content, details):
+        origin = details['Sender']
+        logger.info("Received message from'{}'".format(origin))
+        self._write_log(origin, self._format_log(content, from_me=True))
+
+    def _write_log(self, phone_number, log):
+        with open(self._get_log_path(phone_number), 'a+') as log_file:
+            log_file.write(log)
+
+    def _get_log_path(self, phone_number):
+        file_name = "{}.txt".format(phone_number)
+        return os.path.join(self.folder, file_name)
+
+    @staticmethod
+    def _format_log(content, from_me=True):
+        start_char = '>' if from_me else '<'
+        return "{prefix}\t{msg}\n".format(prefix=start_char, msg=content)
+
+
+def main():
+    DBusGMainLoop(set_as_default=True)  # has to be called first
+    ofono = OfonoBridge()
+    try:
+        ofono.start()
+        mainloop = GLib.MainLoop()  # todo : own thread
+        mainloop.run()
+    except KeyboardInterrupt:
+        logger.info("Caught CTRL-C:exiting without powering off...")
+    except AttributeError:
```

Comment:
`logger.exception` instead of just `error`, and make it a catch-all (as in, `except:` instead of catching AttributeError)? Would be great to have exception information in the logs, as well as protect from future errors.